instance/tests/models/test_openedx_database_mixins.py

```diff
@@ -566,6 +598,30 @@ def test__get_main_database_url(self):
             "mongodb://test:test@test.opencraft.hosting"
         )
 
+    @patch('instance.models.mixins.database.MongoDBInstanceMixin._get_main_database_url')
+    @patch('instance.models.mixins.database.pymongo.MongoClient', autospec=True)
+    def test_deprovision_mongo(self, mock_mongo_client_cls, mock_get_main_db_url):
+        """
+        Test deprovision_mongo calls drop_database.
+        """
+        self.instance = OpenEdXInstanceFactory()
+        self.instance.mongo_provisioned = True
+        self.instance.deprovision_mongo()
+        for database in self.instance.mongo_database_names:
+            mock_mongo_client_cls().drop_database.assert_any_call(database)
+
+    @patch('instance.models.mixins.database.MongoDBInstanceMixin._get_main_database_url')
+    @patch('instance.models.mixins.database.pymongo.MongoClient', autospec=True)
+    def test_ignore_errors_deprovision_mongo(self, mock_mongo_client_cls, mock_get_main_db_url):
+        """
+        Test mongo is set as deprovision when ignoring errors.
+        """
+        self.instance = OpenEdXInstanceFactory()
+        self.instance.mongo_provisioned = True
+        mock_mongo_client_cls().drop_database = Mock(side_effect=PyMongoError())
```

Comment:
Why don't we just patch this method directly?

Reply:
Ah, that's true. Let me modify this. Thanks